src/mlpack/methods/ann/activation_functions/spline_function.hpp

```diff
@@ -53,24 +53,26 @@ class SplineFunction
   /**
    * Computes the first derivative of the Spline function.
    *
-   * @param y Input data.
+   * @param x Input activation.
+   * @param y Result of Fn(x).
    * @return f'(x)
    */
-  static double Deriv(const double y)
+  static double Deriv(const double x, const double /* y */)
   {
-    return  2 * y * std::log(1 + y) + std::pow(y, 2) / (1 + y);
+    return  2 * x * std::log(1 + x) + std::pow(x, 2) / (1 + x);
```

Comment:
```suggestion
    return  2 * y / x + std::pow(x, 2) / (1 + x);
```
Save a log?  (On second thought you also have to handle when `x == 0`, so the expression becomes a bit more complex, but you can still save the expensive log.)

Reply:
saving a log is good!